membership-attribute-service/app/models/AccountDetails.scala

```diff
@@ -180,4 +181,27 @@ object AccountDetails {
 
     }
   }
+
+  /**
+   * Note this is a different concept than termEndDate because termEndDate
+   * could be many years in the future. termEndDate models when Zuora will
+   * renew the subscription whilst anniversary indicates when another year
+   * will have passed since user started their subscription.
+   *
+   * @param start beginning of subscription timeline, perhaps customerAcceptanceDate
+   * @param today where we are on the timeline today
+   * @return next anniversary date of the subscription
+   */
+  def anniversary(
+    start: LocalDate,
+    today: LocalDate = LocalDate.now()
+  ): LocalDate = {
+    @tailrec def loop(left: LocalDate): LocalDate = {
+      val right = left.plusYears(1)
+      if (today.isEqual(left)) right
+      else if (today.isAfter(left) && today.isBefore(right)) right
+      else loop(right)
+    }
+    loop(start)
```

Comment:
Is there a reason this wouldn't be sufficient?
```suggestion
    @tailrec def loop(current: LocalDate): LocalDate = {
      val next = current.plusYears(1)
      if (today.isBefore(next)) next
      else loop(next)
    }
    loop(start)
```